backend/data/processors/manual_fixes/MATHFixes.py

```diff
@@ -88,13 +95,85 @@ def MATH_1099(condition):
     """
-    
+
     return {
         "original": condition["original"],
         "processed": "3956 || 3959",
         "warning": "Must be a Co-Op Scholar"
     }
 
+def MATH_2111():
+    """
+    "original": "Prerequisite: MATH1231 or DPST1014 or MATH1241 or MATH1251 each with a mark of at least 70; Excluded: MATH2019, MATH2049, MATH2069, MATH2100, MATH2110.<br/><br/>",
+    "processed": "MATH1231 || DPST1014 || MATH1241 || MATH1251 each with 70GRADE"
+    """
+
+    COURSES["MATH2111"]["exclusions"]["MATH2019"] = 1
+    COURSES["MATH2111"]["exclusions"]["MATH2049"] = 1
+    COURSES["MATH2111"]["exclusions"]["MATH2069"] = 1
+    COURSES["MATH2111"]["exclusions"]["MATH2100"] = 1
+    COURSES["MATH2111"]["exclusions"]["MATH2110"] = 1
+
+    return "70GRADE in MATH1231 || 70GRADE in DPST1014 || 70GRADE in MATH1241 || 70GRADE in MATH1251"
+
+def MATH_2221():
+    """
+    "original": "Prerequisite: MATH1231 or DPST1014 or MATH1241 or MATH1251 each with a mark of 70.<br/><br/>",
+    "processed": "MATH1231 || DPST1014 || MATH1241 || MATH1251 each with 70GRADE"
+    """
+
+    return "70GRADE in MATH1231 || 70GRADE in DPST1014 || 70GRADE in MATH1241 || 70GRADE in MATH1251"
+
+def MATH_2301():
+    """
+    "original": "Prerequisite: MATH1031 (CR) or MATH1231 or or DPST1014 or MATH1241 or MATH1251. Exclusions: MATH2089, CVEN2002, and CVEN2702<br/><br/>",
+    "processed": "65GRADE in MATH1031 || MATH1231 || or DPST1014 || MATH1241 || MATH1251"
+    """
+
+    COURSES["MATH2301"]["exclusions"]["MATH2089"] = 1
+    COURSES["MATH2301"]["exclusions"]["CVEN2002"] = 1
+    COURSES["MATH2301"]["exclusions"]["CVEN2702"] = 1
+
+    return "65GRADE in MATH1031 || MATH1231 || DPST1014 || MATH1241 || MATH1251"
+
+def MATH_2601():
+    """
+    "original": "Prerequisite: MATH1231 or DPST1014 or MATH1241 or MATH1251 each with a mark of 70 or greater; Exclusion: MATH2099<br/><br/>",
+    "processed": "MATH1231 || DPST1014 || MATH1241 || MATH1251 each with 70GRADE"
+    """
+
+    COURSES["MATH2601"]["exclusions"]["MATH2099"] = 1
+
+    return "70GRADE in MATH1231 || 70GRADE in DPST1014 || 70GRADE in MATH1241 || 70GRADE in MATH1251"
+
+def MATH_2621():
+    """
+    "original": "Prerequisite: MATH1231 or DPST1014 or MATH1241 or MATH1251 each with a mark of at least 70; Exclusion: MATH2069<br/><br/>",
+    "processed": "MATH1231 || DPST1014 || MATH1241 || MATH1251 each with 70GRADE"
+    """
+
+    COURSES["MATH2621"]["exclusions"]["MATH2069"] = 1
+
+    return "70GRADE in MATH1231 || 70GRADE in DPST1014 || 70GRADE in MATH1241 || 70GRADE in MATH1251"
+
+def MATH_2701():
+    """
+    "original": "Prerequisite: MATH1231 or DPST1014 or MATH1241 or MATH1251 with at least a CR, enrolment in an advanced maths or advanced science program<br/><br/>",
+    "processed": "MATH1231 || DPST1014 || MATH1241 || MATH1251 with at least a CR || enrolment in an advanced maths || advanced science program"
+    """
+    # TODO: JOEL: Got these codes from these searches under 'program':
```

Comment:
We actually have  a way to do a query very similar to this algorithmically. For both maths and advanced science, you can add a new clause in https://github.com/csesoc/Circles/blob/c54e26737287f22a7faa0688828bf45ea6ffda7f/backend/algorithms/cache/cache.py#L124
and label them MATH# and ASCI# respectively.

Reply:
You reckon label advanced math as "MATH#"? Might that get confusing with regular maths programs vs advanced maths? I've labelled with "MATH#" for now but can change if needed